clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java

```diff
@@ -73,9 +74,21 @@ public static TachyonConf loadFromHadoopConfiguration(Configuration source) {
         LOG.error("Unable to load TachyonConf from Hadoop configuration", e);
         throw new RuntimeException(e);
       }
-
-      return tachyonConfProperties != null ? new TachyonConf(tachyonConfProperties) : null;
     }
-    return null;
+    if (tachyonConfProperties == null) {
+      tachyonConfProperties = new Properties();
+    }
+    // Load any Tachyon configuration parameters existing in the Hadoop configuration.
+    for (Map.Entry<String, String> entry : source) {
+      String propertyName = entry.getKey();
```

Comment:
I'm not sure, but I think the tricky part might be that this issue shows up when hadoop command line is remote from the tachyon client, so the unit test probably won't be able to exercise that scenario.